ravenframework/utils/randomUtils.py

```diff
@@ -85,11 +87,101 @@ def testSampling(self, n=1e5,engine=None):
     stdev = np.std(samples)
     return mean,stdev
 
+
+class CrowRNG:
+  """ Wraps crow RandomClass to make it serializable """
+  def __init__(self, engine=None, seed=None):
+    """
+      Constructor
+      @ In, engine, RandomClass, optional, will wrap the given engine if provided, otherwise a new engine is created
+      @ In, reseed_copies, bool, whether or not recovered copies of the class should use a new seed or not
+      @ Out, None
+    """
+    if engine is None:
+      self._engine = findCrowModule('randomENG').RandomClass()
+    elif isinstance(engine, findCrowModule('randomENG').RandomClass):
+      self._engine = engine
+    else:
+      raise TypeError(f'Object of unknown type {type(engine)} cannot be wrapped by CrowRNG class!')
+
+    if seed is not None:
+      self._seed = abs(int(seed))
+      self._engine.seed(self._seed)
+    else:
+      self._seed = self._engine.get_rng_seed()
+
+  def __getstate__(self):
+    """
+      Get state for serialization
+      @ In, None
+      @ Out, d, dict, object instance state
+    """
+    d = copy.copy(self.__dict__)
+    eng = d.pop('_engine')  # remove RNG engine from class instance
+    d['count'] = eng.get_rng_state()
+    return d
+
+  def __setstate__(self, d):
+    """
+      Set object instance state
+      @ In, d, dict, object state
+      @ Out, None
+    """
+    count = d.pop('count')
+    self.__dict__.update(d)
+    self._engine = findCrowModule('randomENG').RandomClass()  # reinstantiate RNG engine
+    self._engine.seed(self._seed)
+    self._engine.forward_seed(count)
+
+  def seed(self, value):
+    """
+      Wrapper for RandomClass.seed()
+      @ In, value, int, RNG seed
+      @ Out, None
+    """
+    self._seed = abs(int(value))
+    self._engine.seed(self._seed)  # takes unsigned long
+
+  def random(self):
+    """
+      Wrapper for RandomClass.random()
+      @ In, None
+      @ Out, float, random number from RNG engine
+    """
+    return self._engine.random()  # returns double
+
+  def get_rng_state(self):
+    """
+      Wrapper for RandomClass.get_rng_state()
+      @ In, None
+      @ Out, int, RNG state
+    """
+    return self._engine.get_rng_state()  # returns int
+
+  def forward_seed(self, counts):
+    """
+      Wrapper for RandomClass.forward_seed()
+      @ In, counts, int, number of random states to progress
+      @ Out, None
+    """
+    self._engine.forward_seed(counts)  # takes unsigned int
+
+  def get_rng_seed(self):
+    """
+      Wrapper for RandomClass.get_rng_seed()
+      @ In, None
+      @ Out, int, RNG seed value
+    """
+    val = self._engine.get_rng_seed()  # returns int
+    self._seed = abs(int(val))
+    return self._seed
```

Comment:
Completely missed those. @j-bryan You can make an additional PR for those changes.

Reply:
@wangcj05 @dylanjm The function names in the wrapper class replicate the the crow RandomClass function names to keep the interface consistent. Changing those to camel case would likely require significant refactoring since those functions are used directly through RAVEN. Also, the remaining variable names in the wrapper class appear to follow the code standards, using camel case with one leading underscore in the case of protected variables. Am I misunderstanding that part of the style guide?

Reply:
@j-bryan Here are what I found: 
<img width="992" alt="image" src="https://user-images.githubusercontent.com/7321071/176024794-e4424c60-2912-4f1f-b778-cec0909222f9.png">

I would suggest to change all of them

Reply:
FYI, using one leading underscore is good to us.